resolver_test.go

```diff
@@ -165,3 +167,104 @@ func TestDNSIPQuery(t *testing.T) {
 	w.ClearResponse()
 
 }
+
+var requests = 0
+
+func handleTestDNSRequestServFailOnce(w dns.ResponseWriter, r *dns.Msg) {
+	m := new(dns.Msg)
+	m.SetReply(r)
+	m.Compress = false
+	if requests == 0 {
+		m.SetRcode(r, dns.RcodeServerFailure)
+	}
+	requests = requests + 1
+	w.WriteMsg(m)
+}
+
+func waitForLocalDNSServer(t *testing.T) {
+	retries := 0
+	maxRetries := 10
+
+	for retries < maxRetries {
+		t.Log("Try connecting to DNS server ...")
+		// this test and retry mechanism only works for TCP. With UDP there is no
+		// connection and the test becomes inaccurate leading to unpredictable results
+		tconn, err := net.DialTimeout("tcp", "127.0.0.1:53", 10*time.Second)
+		retries = retries + 1
+		if tconn != nil {
+			defer tconn.Close()
+		}
+		if err != nil {
```

Comment:
Can we club both if together ? it will help readability.  Else block seems to be more. Just a suggestion .
LIke

Reply:
LGTM